torch/onnx/__init__.py

```diff
@@ -1,57 +1,84 @@
+    symbolic_helper,
+    symbolic_opset7,
+    symbolic_opset8,
+    symbolic_opset9,
+    symbolic_opset10,
+    symbolic_opset11,
+    symbolic_opset12,
+    symbolic_opset13,
+    symbolic_opset14,
+    symbolic_opset15,
+    symbolic_opset16,
+    symbolic_registry,
+    utils,
+)
+from ._constants import ONNX_ARCHIVE_MODEL_PROTO_NAME
+from ._exporter_states import ExportTypes, SymbolicContext
+from .errors import CheckerError  # Backwards compatibility
+
+__all__ = [
+    # Modules
+    "symbolic_helper",
+    "symbolic_registry",
+    "utils",
+    "errors",
+    # All opsets
+    "symbolic_caffe2",
+    "symbolic_opset7",
+    "symbolic_opset8",
+    "symbolic_opset9",
+    "symbolic_opset10",
+    "symbolic_opset11",
+    "symbolic_opset12",
+    "symbolic_opset13",
+    "symbolic_opset14",
+    "symbolic_opset15",
+    "symbolic_opset16",
+    # Enums
+    "OperatorExportTypes",
+    "TrainingMode",
+    # Public functions
+    "export",
+    "export_to_pretty_string",
+    "is_in_onnx_export",
+    "select_model_mode_for_export",
+    "register_custom_op_symbolic",
+    "unregister_custom_op_symbolic",
+    "disable_log",
+    "enable_log",
+    "is_onnx_log_enabled",
+    "log",
+    "set_log_stream",
+    # Errors
+    "CheckerError",  # Backwards compatibility
+]
 
 producer_name = "pytorch"
-producer_version = _C._onnx.PRODUCER_VERSION
-
-
-class ExportTypes:
-    r"""Specifies how the ONNX model is stored."""
-
-    PROTOBUF_FILE = "Saves model in the specified protobuf file."
-    ZIP_ARCHIVE = "Saves model in the specified ZIP file (uncompressed)."
-    COMPRESSED_ZIP_ARCHIVE = "Saves model in the specified ZIP file (compressed)."
-    DIRECTORY = "Saves model in the specified folder."
-
-
-class CheckerError(Exception):
-    r"""Raised when ONNX checker detects an invalid model."""
-
-    pass
-
-
-class SymbolicContext:
-    r"""Provides extra context for symbolic functions.
-
-    Args:
-        params_dict (Dict[str, _C.IValue]): Mapping from graph initializer name to IValue.
-        env (Dict[_C.Value, _C.Value]): Mapping from Torch domain graph Value to ONNX domain graph Value.
-        cur_node (_C.Node): Current node being converted to ONNX domain.
-        onnx_block (_C.Block): Current ONNX block that converted nodes are being appended to.
-    """
-
-    def __init__(self, params_dict, env, cur_node, onnx_block):
-        self.params_dict: Dict[str, _C.IValue] = params_dict
-        self.env: Dict[_C.Value, _C.Value] = env
-        # Current node that is being converted.
-        self.cur_node: _C.Node = cur_node
-        # Current onnx block that converted nodes are being appended to.
-        self.onnx_block: _C.Block = onnx_block
+producer_version = _C_onnx.PRODUCER_VERSION
 
 
 def _export(*args, **kwargs):
-    from torch.onnx import utils
-
-    result = utils._export(*args, **kwargs)
-    return result
+    warnings.warn(
+        "`torch.onnx._export` is deprecated. Please use `export` instead.",
```

Comment:
I'm not sure about this. I agree with the intention. However, we may need to propose the alternative along with message of deprecation.

Reply:
The message proposes `export`. Are there cases when we needed to use `torch.onnx._export`? Internally we can always use `utils._export`

Reply:
Users that depend on additional arguments in `_export` that is not in `export`. I'm ok with this. Let's add this to the description.

Reply:
I updated the pr description. Should we consider adding additional arguments to ExportOptions and support them there? This way we have a unified api for exporting.